lib/services/role.go

```diff
@@ -629,14 +629,17 @@ func ApplyValueTraits(val string, traits map[string][]string) ([]string, error)
 				return trace.BadParameter("unsupported variable %q", name)
 			}
 		}
-		// TODO: return a not found error if the variable namespace is not
-		// the namespace of `traits`.
-		// If e.g. the `traits` belong to the "internal" namespace (as the
-		// validation above suggests), and "foo" is a key in `traits`, then
-		// "external.foo" will return the value of "internal.foo". This is
-		// incorrect, and a not found error should be returned instead.
-		// This would be similar to the var validation done in getPAMConfig
-		// (lib/srv/ctx.go).
+		// The "external" trait namespace is explicitly allowed to reference
+		// "internal" traits listed above. This is for multiple reasons:
```

Comment:
Do we have test coverage for this? It has caused strangeness before, so I think it's better to make sure it's covered to avoid eventual regressions.

Reply:
added test coverage in b987047dba17bd98283b606d1f774693608c73c1